tests/test_core.py

```diff
@@ -36,64 +38,111 @@ def test_inspect_dummy_execs():
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            } ,
+            False
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_2',
             Path(__file__).parent / 'cases' / 'executor_2.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_3',
             Path(__file__).parent / 'cases' / 'executor_3.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_4',
             Path(__file__).parent / 'cases' / 'executor_4.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_5',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_6',
             None,
+            {},
+            True
+        ),
+        (
+            Path(__file__).parent / 'cases' / 'executor_7',
+            None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            },
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested',
             Path(__file__).parent / 'cases' / 'nested.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_2',
             None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            },
+            False
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_3',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_4',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_5',
             None,
+            {},
+            True
         ),
     ],
 )
-def test_get_executor_args(package_path, expected_path):
+def test_get_executor_args(package_path, expected_path, build_args_envs, dry_run):
     if expected_path:
         with open(expected_path, 'r') as fp:
             expected_executor = ExecutorModel(**json.loads(fp.read()))
-            executor = core.normalize(package_path, dry_run=True)
+            executor = core.normalize(package_path, build_args_envs=build_args_envs, dry_run=dry_run)
             executor.hubble_score_metrics = expected_executor.hubble_score_metrics
             executor.filepath = expected_executor.filepath
             assert executor == expected_executor
     else:
-        core.normalize(package_path, dry_run=True)
+        core.normalize(package_path, build_args_envs=build_args_envs, dry_run=dry_run)
+    
+    dockerfilePath = Path(package_path / 'Dockerfile') 
+    if dry_run is False and dockerfilePath.exists():
```

Comment:
Make sure the `assert` is easy to understand. In this case, you should add another expected modified Dockerfile and compare these two.